extensions/amp-react-img/0.1/amp-react-img.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {devAssert} from '../../../src/log';
+import {requireExternal} from '../../../src/module';
+
+/**
+ * @param {!Object<string, *>} props
+ * @param {!Array<string>} keys
+ * @return {!Object<string, *>}
+ */
+function pick(props, keys) {
+  const out = {};
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    const value = props[key];
+    if (value !== undefined) {
+      out[key] = value;
+    }
+  }
+
+  return out;
+}
+
+const ATTRIBUTES_TO_PROPAGATE = [
+  'alt',
+  'title',
+  'referrerpolicy',
+  'aria-label',
+  'aria-describedby',
+  'aria-labelledby',
+  'srcset',
+  'src',
+  'sizes',
+];
+
+AMP.extension('amp-react-img', '0.1', AMP => {
+  const react = requireExternal('react');
+  const {render} = requireExternal('react-dom');
+
+  /**
+   * We'll implement all our new extensions as React/Preact Components (TBD).
+   * They're true Components, not AmpElements/Amp.BaseElements.
+   */
+  class AmpImg extends react.Component {
+    /**
+     * @param {!Object} props
+     */
+    constructor(props) {
+      super(props);
+
+      this.state = {prerender: true};
```

Comment:
Definitely subject to bikeshedding, but imho, this should be something like "loading=auto|manual", since there might be other reasons to block rendering, besides prerendering.

Reply:
Yup, I'm not tied to `prerender` at all, it just the first name I thought of.